cmd/githubgen/main.go

```diff
@@ -0,0 +1,204 @@
+# Path separator and minimum of 1 space between component path and owners is
+# important for validation steps
+#
+
+* @open-telemetry/collector-contrib-approvers
+`
+
+const unmaintainedHeader = `
+
+## UNMAINTAINED components
+## The Github issue template generation code needs this to generate the corresponding labels.
+
+`
+
+const allowlistHeader = `#####################################################
+#
+# List of components in OpenTelemetry Collector Contrib
+# waiting on owners to be assigned
+#
+#####################################################
+#
+# Learn about membership in OpenTelemetry community:
+#  https://github.com/open-telemetry/community/blob/main/community-membership.md
+#
+#
+# Learn about CODEOWNERS file format:
+#  https://help.github.com/en/articles/about-code-owners
+#
+
+## 
+# NOTE: New components MUST have a codeowner. Add new components to the CODEOWNERS file instead of here.
+##
+
+## COMMON & SHARED components
+internal/common
+
+`
+
+// Generates files specific to Github according to status metadata:
+// .github/CODEOWNERS
+// .github/ALLOWLIST
+func main() {
+	flag.Parse()
+	folder := flag.Arg(0)
+	if err := run(folder); err != nil {
+		log.Fatal(err)
+	}
+}
+
+type Codeowners struct {
+	// Active codeowners
+	Active []string `mapstructure:"active"`
+	// Emeritus codeowners
+	Emeritus []string `mapstructure:"emeritus"`
+}
+type Status struct {
+	Stability     map[string][]string `mapstructure:"stability"`
+	Distributions []string            `mapstructure:"distributions"`
+	Class         string              `mapstructure:"class"`
+	Warnings      []string            `mapstructure:"warnings"`
+	Codeowners    *Codeowners         `mapstructure:"codeowners"`
+}
+type metadata struct {
+	// Type of the component.
+	Type string `mapstructure:"type"`
+	// Type of the parent component (applicable to subcomponents).
+	Parent string `mapstructure:"parent"`
+	// Status information for the component.
+	Status *Status `mapstructure:"status"`
+}
+
+func loadMetadata(filePath string) (metadata, error) {
+	cp, err := fileprovider.New().Retrieve(context.Background(), "file:"+filePath, nil)
+	if err != nil {
+		return metadata{}, err
+	}
+
+	conf, err := cp.AsConf()
+	if err != nil {
+		return metadata{}, err
+	}
+
+	md := metadata{}
+	if err := conf.Unmarshal(&md); err != nil {
+		return md, err
+	}
+
+	return md, nil
+}
+
+func run(folder string) error {
+	components := map[string]metadata{}
+	foldersList := []string{}
+	maxLength := 0
+	err := filepath.Walk(folder, func(path string, info fs.FileInfo, err error) error {
+		if info.Name() == "metadata.yaml" {
+			m, err := loadMetadata(path)
+			if err != nil {
+				return err
+			}
+			if m.Status == nil {
+				return nil
+			}
+			key := filepath.Dir(path) + "/"
+			components[key] = m
+			foldersList = append(foldersList, key)
+			for stability := range m.Status.Stability {
+				if stability == "unmaintained" {
```

Comment:
The plain string "unmaintained" is used in a couple places, might be nice to make it a variable for re-use.

Reply:
added, thanks :)